lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -241,78 +239,107 @@ private static Domain getDomain(
         }
 
         if (type.equals(BIGINT) || type.equals(INTEGER) || type.equals(DATE) || type.equals(SMALLINT) || type.equals(TINYINT)) {
-            ParquetLongStatistics longStatistics = fromNumber((Number) min, (Number) max);
-            if (isStatisticsOverflow(type, longStatistics)) {
-                return Domain.create(ValueSet.all(type), hasNullValue);
+            List<Range> ranges = new ArrayList<>();
+            for (int i = 0; i < minimums.size(); i++) {
+                long min = longExact((Number) minimums.get(i));
+                long max = longExact((Number) maximums.get(i));
+                if (isStatisticsOverflow(type, min, max)) {
+                    return Domain.create(ValueSet.all(type), hasNullValue);
+                }
+
+                ranges.add(Range.range(type, min, true, max, true));
```

Comment:
Going through single-element Ranges isn't necessarily the most optimal way, and certainly incurs memory copying when final SortedRangeSet is constructed. Did you try putting ranges in a flat block to avoid that?

Reply:
It's possible it's better, but there's no API to do that currently and it would be a bigger change. This basically restores the old way of constructing the domains.